R/format_data.R

```diff
@@ -2414,6 +2414,314 @@ fmt_datetime <- function(data,
   )
 }
 
+#' Format values as time durations
+#'
+#' @description
+#' Format input values to time durations whether those input values are numbers
+#' or of the `difftime` class. We can specify which time units any numeric input
+#' values have (as days, hours, minutes, or seconds) and the output can be
+#' customized with a duration style (corresponding to short, long, and
+#' colon-separated forms) and a choice of output units from weeks to seconds.
+#'
+#' @details
+#' Targeting of values is done through `columns` and additionally by `rows` (if
+#' nothing is provided for `rows` then entire columns are selected). Conditional
+#' formatting is possible by providing a conditional expression to the `rows`
+#' argument. See the *Arguments* section for more information on this.
+#'
+#' @inheritParams fmt_number
+#' @param input_values If one or more selected columns contains numeric values,
+#'   the keyword provided for `input_values` will determine how those values are
+#'   interpreted in terms of a single duration type. By default, **gt** assumes
+#'   that numeric values are representative of `"days"` but this can be changed
+#'   to either of `"hours"`, `"minutes"`, or `"seconds"`.
+#' @param output_units The output values can be handled by one of two keywords
+#'   (`"auto"`, the default, or `"all"`) or a vector of the following time
+#'   parts: `"weeks"`, `"days"`, `"hours"`, `"minutes"`, or `"seconds"`.
+#' @param duration_style A choice of three formatting styles for the output
+#'   duration values. With `"narrow"` (the default style), durations will be
+#'   formatted with single letter time-part units (e.g., 1.35 days will be
+#'   styled as `"1d 8h 24m`). With `"wide"`, this example value will be expanded
+#'   to `"1 day 8 hours 24 minutes"` after formatting. The `"colon-sep"` style
+#'   of formatting will separate all values with a colon and will not provide
+#'   any time-part units, so the output of 1.35 days will be `"1:8:24"`.
+#' @param drop_zero_parts An option to drop leading and internal values for
+#'   time parts if they are zero. By default this is `TRUE` and duration values
+#'   that might otherwise be formatted as `"0w 1d 0h 4m 19s"` with
+#'   `drop_zero_parts = FALSE` are instead displayed as `"1d 4m 19s"`.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @family Format Data
+#' @section Function ID:
+#' 3-11
+#'
+#' @import rlang
+#' @export
+fmt_duration <- function(
+    data,
+    columns,
+    rows = everything(),
+    input_values = c("days", "hours", "minutes", "seconds"),
+    output_units = "auto",
+    duration_style = c("narrow", "wide", "colon-sep"),
+    drop_zero_parts = TRUE,
+    pattern = "{x}"
+) {
+  input_values <- match.arg(input_values)
```

Comment:
Instead of using days as a default, throw an error if one or more columns are numeric and input_values is not provided